lib/palindrome_permutation.rb

```diff
@@ -1,4 +1,14 @@
 
 def palindrome_permutation?(string)
-  raise NotImplementedError, "palindrome_permutation? not implemented"
+  string_array = string.split("")
+  hash_of_letters = {}
+  string_array.each do |letter|
+    if hash_of_letters.keys.include?(letter)
```

Comment:
Remember that `hash_of_letters.keys` returns an array and `.include?` is an O(n) operation.  This would be **much** better as:

```suggestion
    if hash_of_letters[letter]
```

This way you take advantage of the O(1) lookup time of a hash.